### PR TITLE
Support path objs in components

### PR DESCRIPTION
**Pull request by sunilkumardash9**
# Description

Added support for Path objects in components including gr.Video, gr.Gallery, gr.chatbot. Audio, image and model3d are working fine with Path. 

Closes: #4496 

# Checklist:

- [🗸] I have performed a self-review of my code
- [ ] I have added a summary of my change to the CHANGELOG.md
- [ ] My code follows the style guidelines of this project
- [ ] I have commented on my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ 🗸] New and existing unit tests pass locally with my changes

# A note about the CHANGELOG

Hello 👋 and thank you for contributing to Gradio!

All pull requests must update the change log located in CHANGELOG.md unless the pull request is labelled with the "no-changelog-update" label.

Please add a summary of the change to the Upcoming Release > Full Changelog section of the CHANGELOG.md file and include
a link to the PR (formatted in markdown) and a link to your GitHub profile (if you like). For example, "* Added a cool new feature by `[@myusername](link-to-your-github-profile)` in `[PR 11111](https://github.com/gradio-app/gradio/pull/11111)`".

If you would like to elaborate on your change further, feel free to include a longer explanation in the other sections.
If you would like an image/gif/video showcasing your feature, it may be best to edit the CHANGELOG file using the 
GitHub web UI since that lets you upload files directly via drag-and-drop.

### CODE REVIEW

**Comment by abidlabs** [comment]
Hi @sunilkumardash9 thanks for the contribution! Can you update the types in the docstrings to designate that `pathlib.Path`s are now supported? And you can kindly add some tests in `test/test_components.py` to test support of `pathlib.Path`s?

**Comment by sunilkumardash9** [comment]
> Hi @sunilkumardash9 thanks for the contribution! Can you update the types in the docstrings to designate that `pathlib.Path`s are now supported? And you can kindly add some tests in `test/test_components.py` to test support of `pathlib.Path`s?

Should we test this inside each test component class or a separate test? And I would like to know if instantiating a component with pathlib.Path and asserting them should be enough or we need some other tests.

**Comment by abidlabs** [comment]
> Should we test this inside each test component class or a separate test? 

Inside each test component class 

> And I would like to know if instantiating a component with pathlib.Path and asserting them should be enough or we need some other tests.

Would you be able to write tests instantiating each component with `pathlib.Path` and then confirming that the post-processed value is what you expect? You can reference this test as an example:

https://github.com/gradio-app/gradio/blob/9e13cf3890de3024c729031f763767c82d3b7d63/test/test_components.py#L1862

Thanks @sunilkumardash9!

**Comment by sunilkumardash9** [comment]
@abidlabs Thanks for the input. Wrote tests on a similar line. Do we need to test other components (Audio, Image, etc.) for pathlib.Path?

**Comment by abidlabs** [comment]
@sunilkumardash9 if would like to add tests and update the docstrings for the other components, that would be much appreciated! If not, we can keep this PR scoped to these components.

**Comment by abidlabs** [comment]
Did some cleanup + credited you in the changelog, but otherwise looks good to go! Thanks @sunilkumardash9 for the contribution

File: test/test_components.py [hunk 1]
```diff
@@ -2167,6 +2208,14 @@ def test_gallery(self, mock_uuid):
             data_restored = [d[0]["data"] for d in data_restored]
             assert sorted(data) == sorted(data_restored)
 
+        # test if the Gallery component supports pathlib.Path objects
```
Comment (abidlabs) [inline test/test_components.py:2211]:
```suggestion
```

File: test/test_components.py [hunk 2]
```diff
@@ -1375,6 +1375,44 @@ def test_component_functions(self):
             )
         ).endswith(".mp4")
 
+        # test if the video component supports pathlib.Path objects
```
Comment (abidlabs) [inline test/test_components.py:1378]:
```suggestion
```

**Comment by abidlabs** [comment]
Added docstrings, typing, and tests for the remaining components as well. Will merge in after CI

**Pull request merged**